src/xlsx/conversion/figure_conversion.ts

```diff
@@ -12,31 +12,50 @@ import {
   ChartDefinition,
   ExcelChartDefinition,
   ExcelChartTrendConfiguration,
-  FigureData,
   HeaderIndex,
   PixelPosition,
+  SheetData,
   TrendConfiguration,
 } from "../../types";
-import { AnchorOffset } from "../../types/figure";
-import { ExcelImage } from "../../types/image";
+import { AnchorOffset, Figure } from "../../types/figure";
+import { ExcelImage, Image } from "../../types/image";
 import { XLSXFigure, XLSXWorksheet } from "../../types/xlsx";
 import { convertEMUToDotValue, getColPosition, getRowPosition } from "../helpers/content_helpers";
 import { XLSXFigureAnchor } from "./../../types/xlsx";
 import { convertColor } from "./color_conversion";
 import { EXCEL_TO_SPREADSHEET_TRENDLINE_TYPE_MAPPING } from "./conversion_maps";
 
-export function convertFigures(sheetData: XLSXWorksheet): FigureData<any>[] {
+export function convertFigures(sheetData: XLSXWorksheet): {
+  figures: SheetData["figures"];
+  images: SheetData["images"];
+  charts: SheetData["charts"];
+} {
+  const figures: SheetData["figures"] = [];
+  const images: SheetData["images"] = {};
+  const charts: SheetData["charts"] = {};
+
   let id = 1;
-  return sheetData.figures
+  const figs = sheetData.figures
     .map((figure) => convertFigure(figure, (id++).toString(), sheetData))
     .filter(isDefined);
+
+  for (const { figure, chart, image } of figs) {
+    figures.push(figure);
+    if (chart) {
+      charts[figure.id] = { figureId: figure.id, chart };
+    } else if (image) {
```

Comment:
There's unnecessary complexity that could be improved here. There are 2 *"if chart else if image ..."* (here and in `convertFigure`)  while the code could be reorganized to have only one.